packages/block-library/src/template-part/edit/placeholder/index.js

```diff
@@ -59,13 +59,19 @@ export default function TemplatePartPlaceholder( {
 			startingBlocks = [],
 			title = __( 'Untitled Template Part' )
 		) => {
+			// Currently template parts only allow latin chars.
+			// Fallback slug will receive suffix by default.
+			const cleanSlug =
+				kebabCase( title ).replace( /[^\w-]+/g, '' ) ||
+				'wp-custom-part';
+
 			// If we have `area` set from block attributes, means an exposed
 			// block variation was inserted. So add this prop to the template
 			// part entity on creation. Afterwards remove `area` value from
 			// block attributes.
 			const record = {
 				title,
-				slug: kebabCase( title ),
+				slug: cleanSlug,
```

Comment:
Since, https://github.com/WordPress/gutenberg/pull/39033 landed, maybe try rebasing and using `cleanSlug` from the url package?

Reply:
I also need to update REST API [schema](https://github.com/WordPress/gutenberg/blob/1df72fc1b3d628b2bc7b13bed74f6b25ab44f82b/lib/compat/wordpress-5.9/class-gutenberg-rest-templates-controller.php#L597-L604), but first want to make sure only Latin chars limitation isn't intentional.